### DISCUSSION TITLE
Reticulum Compilation Error

### DISCUSSION THREAD

**Discussion by NagarajuSamudrala** [mozilla/hubs#4990] [General Questions & Discussion] [answered]
Hello,

When compiling Reticulum I am getting compilation error when running the last step - "iex -S mix phx.server".

Error is as follows ..

![image](https://user-images.githubusercontent.com/97037908/147942511-f935788a-067f-457e-9e79-fa338da8f06c.png)

[debug] QUERY OK db=1.3ms queue=0.5ms idle=72.0ms
INSERT INTO coturn.turn_secret (realm, value, inserted_at, updated_at) values ($1, $2, now(), now()) ["ret", "f0aca77768277e4b54f7c91c46b96ad7"]
[debug] QUERY OK db=1.0ms queue=0.6ms idle=24.0ms
**DELETE FROM coturn.turn_secret WHERE inserted_at < now() - interval '15 minutes' []
[notice] Application ret exited: Ret.Application.start(:normal, []) returned an error: shutdown: failed to start child: RetWeb.Endpoint**
    ** (EXIT) shutdown: failed to start child: Phoenix.PubSub.PG2
        ** (EXIT) shutdown: failed to start child: Phoenix.PubSub.PG2Server
            ** (EXIT) an exception was raised:
                ** (UndefinedFunctionError) function :pg2.create/1 is undefined or private
                    (kernel 8.1.3) :pg2.create({:phx, Ret.PubSub})
                    (phoenix_pubsub 1.1.2) lib/phoenix/pubsub/pg2_server.ex:43: Phoenix.PubSub.PG2Server.init/1
                    (stdlib 3.16.1) gen_server.erl:423: :gen_server.init_it/2
                    (stdlib 3.16.1) gen_server.erl:390: :gen_server.init_it/6
                    (stdlib 3.16.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3

Need your help in understanding this error .. what it means and how to fix it.

Thanks,
Nagaraju Samudrala

**Comment by NagarajuSamudrala** [comment, marked as answer]
Hello,

Once downgrading the phoenix server, elixir and reticulum, I am able to run reticulum

Thanks,
Nagaraju Samudrala

**Reply by ipeakermit** [reply]
What version/s did you downgrade to?

**Comment by jreoka** [comment]
Getting this same error...
